ir/function.cpp

```diff
@@ -48,6 +48,54 @@ JumpInstr::it_helper BasicBlock::targets() const {
   return {};
 }
 
+void BasicBlock::prunePhiSrc(const BasicBlock *bb) {
```

Comment:
Nit: can we use `removePhiSrc` instead? I'm not sure whether prune is a good name, but `remove` is clear IMO